packages/mobile/src/account/profileInfo.ts

```diff
@@ -0,0 +1,317 @@
+  if (isAlreadyUploaded) {
+    return
+  }
+  try {
+    yield call(unlockDEK, true)
+    // yield call(addMetadataClaim)
+    yield call(uploadNameAndPicture)
+
+    yield put(profileUploaded())
+  } catch (e) {
+    Logger.error(TAG + '@uploadProfileInfo', 'Error uploading profile', e)
+  }
+}
+
+// TODO: make metadata claim when registering account info, and use fetched metadata url when reading data
+
+// export function* addMetadataClaim() {
+//   try {
+//     const contractKit = yield call(getContractKit)
+//     const account = yield select(currentAccountSelector)
+//     const metadata = IdentityMetadataWrapper.fromEmpty(account)
+//     yield call(
+//       [metadata, 'addClaim'],
+//       createStorageClaim(BUCKET_URL),
+//       NativeSigner(contractKit.web3.eth.sign, account)
+//     )
+//     Logger.info(TAG + '@addMetadataClaim' + 'created storage claim on chain')
+//     yield call(writeToGCPBucket, metadata.toString(), `${account}/metadata.json`)
+//     Logger.info(TAG + '@addMetadataClaim' + 'uploaded metadata.json')
+//     const accountsWrapper: AccountsWrapper = yield call([
+//       contractKit.contracts,
+//       contractKit.contracts.getAccounts,
+//     ])
+//     const setAccountTx = accountsWrapper.setMetadataURL(`${BUCKET_URL}${account}/metadata.json`)
+//     const context = newTransactionContext(TAG, 'Set metadata URL')
+//     yield call(sendTransaction, setAccountTx.txo, account, context)
+//     Logger.info(TAG + '@addMetadataClaim' + 'set metadata URL')
+//   } catch (error) {
+//     Logger.error(`${TAG}/addMetadataClaim`, 'Could not add metadata claim', error)
+//     throw error
+//   }
+// }
+
+export function* uploadNameAndPicture() {
+  const contractKit = yield call(getContractKit)
+  const account = yield select(currentAccountSelector)
+  const offchainWrapper = new UploadServiceDataWrapper(contractKit, account)
+
+  const name = yield select(nameSelector)
+  const nameAccessor = new PrivateNameAccessor(offchainWrapper)
+  let writeError = yield call([nameAccessor, 'write'], { name }, [])
+  Logger.info(TAG + 'uploadNameAndPicture', 'uploaded profile name')
+
+  const pictureUri = yield select(pictureSelector)
+  if (pictureUri) {
+    const data = yield call(RNFS.readFile, pictureUri, 'base64')
+    const mimeType = extensionToMimeType[pictureUri.split('.').slice(-1)] || 'image/jpeg'
+    const dataURL = getDataURL(mimeType, data)
+    const pictureAccessor = new PrivatePictureAccessor(offchainWrapper)
+    writeError = yield call([pictureAccessor, 'write'], dataURL, [])
+    Logger.info(TAG + 'uploadNameAndPicture', 'uploaded profile picture')
+  }
+
+  if (writeError) {
+    Logger.error(TAG + '@uploadNameAndPicture', writeError)
+    throw Error('Unable to write data')
+  }
+}
+
+// this function gives permission to the recipient to view the user's profile info
+export function* uploadSymmetricKeys(recipientAddresses: string[]) {
```

Comment:
I'd also consider renaming this. References symmetric keys in the function name feels to low level for this API.